worksheet_B/Mandelbrot.cpp

```diff
@@ -18,38 +138,67 @@ int main()
 	for (int pixelY = 0; pixelY < image.height(); pixelY++)
 	{
 		// TODO: Map the y coordinate into the range minY to maxY
-		//double y0 =
+		double y0 = (maxY - minY) * (pixelY / 800.0) + minY;
 
-		for (int pixelX = 0; pixelX < image.width(); pixelX++)
+		for (int pixelX = 0.0; pixelX < image.width(); pixelX++)
 		{
 			// TODO: Map the x coordinate into the range minX to maxX
-			//double x0 =
+			double x0 = (maxX - minX) * (pixelX / 800.0) + minX;
 
 			// TODO: implement the algorithm to colour a single pixel (x0, y0) of the Mandelbrot set fractal
-			// The code below simply fills the screen with random pixels
+			double y = 0.0;
+			double x = 0.0;
+			double iteration = 0.0;
+			double max_iteration = 200;
+
+			while (iteration < max_iteration && x * x + y * y < 2*2 )
+			{
+				double temp_x = x * x - y * y + x0;
+				y = (2 * x * y) + y0;
+				x = temp_x;
+				++iteration;
 
-			// Write the pixel
+			}
+
+			rgb rgb_values;
+
+			if (iteration > 20 && iteration < 180)
+			{
+				// changes outer colour
+				rgb_values = { 100.0, 100.0, 255.0 };
+			}
+			else if (iteration >= max_iteration)
+			{
+				// changes inner colour
+				rgb_values = { 10.0, 10.0, 10.0 };
+			}
+			else
+			{
+				hsv hsv_values = { (iteration - 100) , 50.0, 50.0 };
+				rgb_values = hsv2rgb(hsv_values);
+			}
 			// TODO: change the right-hand side of these three lines to write the desired pixel colour value
-			image(pixelX, pixelY, 0, 0) = rand(); // red component
-			image(pixelX, pixelY, 0, 1) = rand(); // green component
-			image(pixelX, pixelY, 0, 2) = rand(); // blue component
+			image(pixelX, pixelY, 0, 0) = rgb_values.g; // red
```

Comment:
This is unexpected! Worth adding a comment that you are swizzling the RGB values here for a more pleasing aesthetic effect